tests/integration_tests/cpu_tests/bittensor/test_mnist_node.py

```diff
@@ -53,31 +55,37 @@ def setUp(self):
     
     # Test training.
     def test(self):
+        logger.info('Start ... ')
         with self.session:
-            best_loss = math.inf
-            best_accuracy = 0
-            epoch = 0
-            total_epochs = 1
-            while epoch < total_epochs:
-
-                self.train(self.model, epoch)
-                test_loss, accuracy = self.validate(self.model)
-                self.scheduler.step()
-                epoch += 1
-
-                best_accuracy = accuracy if accuracy >= best_accuracy else best_accuracy
-                if test_loss < best_loss:
-                    best_loss = test_loss
-                    logger.info("Best test loss: {}".format(best_loss))
-                    self.session.serve( self.model.deepcopy() )
-
-            assert best_loss <= 0.1
-            assert best_accuracy >= 75 
-            # Neurons should be 3, but we are setting to >= 2 due to the
-            # unpredictability of circleci booting up neurons, some neurons might
-            # start and finish before others even boot up.
-            # TODO: bring back this assert once chain issues are resolved
-            #assert len(self.session.metagraph.neurons()) >= 2
+            Asyncio.start_in_thread(self.start, self.session)
+            Asyncio.run_forever()
+
+    def start(self):
+        # Starts test within the session context and asyncio thread.
+        best_loss = math.inf
+        best_accuracy = 0
+        epoch = 0
+        total_epochs = 1
+        while epoch < total_epochs:
+
+            self.train(self.model, epoch)
+            test_loss, accuracy = self.validate(self.model)
+            self.scheduler.step()
+            epoch += 1
+
+            best_accuracy = accuracy if accuracy >= best_accuracy else best_accuracy
+            if test_loss < best_loss:
+                best_loss = test_loss
+                logger.info("Best test loss: {}".format(best_loss))
+                self.session.serve( self.model.deepcopy() )
+
+        assert best_loss <= 0.1
+        assert best_accuracy >= 75 
+        # Neurons should be 3, but we are setting to >= 2 due to the
+        # unpredictability of circleci booting up neurons, some neurons might
+        # start and finish before others even boot up.
+        # TODO: bring back this assert once chain issues are resolved
```

Comment:
We should add an assert for the number of peers here, we would've caught the tests incorrectly passing if we had done that earlier, no?

Reply:
However, because we are using a shared chain this number is going to be all over the place.